File: system-test/dotnet/SystemTests/Latest/ExternalSystemContractTests/Tax/TaxStubContractTest.cs
using Dsl.Port;

namespace SystemTests.Latest.ExternalSystemContractTests.Tax;

public class TaxStubContractTest : BaseTaxContractTest
{
    protected override ExternalSystemMode? FixedExternalSystemMode => ExternalSystemMode.Stub;

    [Fact]
    public async Task ShouldBeAbleToGetConfiguredTaxRate()
    {
        (await Scenario()
            .Given().Country().WithCode("LALA").WithTaxRate(0.23m)
            .Then().Country("LALA"))
            .HasCountry("LALA")
            .HasTaxRate(0.23m);
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/latest/contract/tax/TaxStubContractTest.java
package com.mycompany.myshop.systemtest.latest.contract.tax;

import com.mycompany.myshop.systemtest.configuration.ExternalSystemMode;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

class TaxStubContractTest extends BaseTaxContractTest {
    @Override
    protected ExternalSystemMode getFixedExternalSystemMode() {
        return ExternalSystemMode.STUB;
    }

    @Test
    void shouldBeAbleToGetConfiguredTaxRate() {
        assertDoesNotThrow(() -> scenario
                .given().country().withCode("LALA").withTaxRate(0.23)
                .then().country("LALA").hasCountry("LALA").hasTaxRate(0.23));
    }
}
